File: app/src/main/java/com/example/gmailish/data/mappers/LabelMapper.java
// LabelMapper.java
package com.example.gmailish.data.mappers;


import com.example.gmailish.data.entity.LabelEntity;

import org.json.JSONObject;

public final class LabelMapper {
    public static LabelEntity fromJson(JSONObject json) {
        return new LabelEntity(
                json.optString("id"),
                json.optString("ownerId"),
                json.optString("name", json.optString("id"))
        );
    }
    private LabelMapper() {}
}

File: app/src/main/java/com/example/gmailish/data/mappers/LabelMapper.kt
package com.example.gmailish.data.mapper

import com.example.gmailish.data.entity.LabelEntity
import org.json.JSONObject

object LabelMapper {
    fun fromJson(json: JSONObject): LabelEntity {
        return LabelEntity(
            id = json.optString("id"),
            ownerId = json.optString("ownerId"),
            name = json.optString("name", json.optString("id"))
        )
    }
}